docs/reference/ccr/use-cases.asciidoc

```diff
@@ -0,0 +1,121 @@
+[role="xpack"]
```

Comment:
I think this level of information is super helpful. I'd be inclined to avoid "use case" (which has a strong application/solution connotation) in favor of something like "Multi-cluster architectures". 

I'm also wondering if this can be incorporated into the overview--it's really information that you are probably interested in *before* you read about the mechanics of the implementation.

Maybe the overview could look something like:

Cross cluster replication
    Replication model
    Multi-cluster architectures
    Replication mechanics
        Initializing followers uses remote recovery
        Replicating a leader relies on soft deletes

Reply:
I waffled on that idea, but I think you're right - integrating that information into the overview makes sense.